src/trix/elements/trix_editor_element.js

```diff
@@ -348,6 +348,8 @@ class LegacyDelegate {
 export default class TrixEditorElement extends HTMLElement {
   static formAssociated = "ElementInternals" in window
 
+  static observedAttributes = [ "initialized" ]
```

Comment:
While I think the naming of this attribute is ultimately an arbitrary implementation detail, I wonder if there are naming conventions that already exist that it could draw inspiration from. For example, the `<turbo-cable-stream-source>` element that `turbo-rails` provides toggles an attribute named "[connected](https://github.com/hotwired/turbo-rails/blob/main/app/javascript/turbo/cable_stream_source_element.js#L35-L41)" which mirrors the `connectedCallback()` naming convention and the [Node.isConnected](https://developer.mozilla.org/en-US/docs/Web/API/Node/isConnected) property.

Reply:
I like that. I renamed it to `connected`, it's removing the attribute on disconnect. I also added an additional check to make sure the element is connected to the dom before resetting the state when the attribute is removed.